packages/core/src/components/menu-category/test/menu-category.ct.ts

```diff
@@ -320,3 +320,52 @@ test('show category if item are focused', async ({ mount, page }) => {
   const dropdown = categoryElement.locator('ix-dropdown');
   await expect(dropdown).toBeVisible();
 });
+
+test('should adjust height when items are added dynamically', async ({
+  mount,
+  page,
+}) => {
+  await mount(`
+    <ix-application>
+      <ix-menu>
+        <ix-menu-category label="Category label">
+          <ix-menu-item>Test</ix-menu-item>
+          <ix-menu-item>Test</ix-menu-item>
+        </ix-menu-category>
+      </ix-menu>
+    </ix-application>
+  `);
+
+  await page.locator('ix-menu-expand-icon').click();
+  await page.locator('ix-menu-category').click();
+  await page.waitForSelector('ix-menu-item:visible');
+
+  const initialHeight: number | undefined = await page
+    .locator('ix-menu-category')
+    .boundingBox()
+    .then((box) => box?.height);
+
+  await page.evaluate(() => {
+    const menuCategory = document.querySelector('ix-menu-category');
+
+    const newMenuItem1 = document.createElement('ix-menu-item');
+    const newMenuItem2 = document.createElement('ix-menu-item');
+
+    newMenuItem1.textContent = 'Dynamically Added';
+    newMenuItem2.textContent = 'Dynamically Added';
+
+    menuCategory?.appendChild(newMenuItem1);
+    menuCategory?.appendChild(newMenuItem2);
+  });
+
+  await page.waitForTimeout(300);
```

Comment:
Using timeouts should be avoided.

Reply:
I just added another timeout. Without them the height measurement is very inaccurate/inconsistent because of the animation that menu-category uses. Do you have another idea?

Reply:
I think smth like this could work: 
```suggestion
  //When the animation is done also the opacity is set to one
    await page.waitForSelector(
    'div.menu-items.menu-items--expanded[style*="opacity: 1"]'
  );
  
  ....
  
  //The second time we could check if the foruth children is visible:
    await page.waitForSelector('ix-menu-category ix-menu-item:nth-child(4)', {
    state: 'visible',
  });

```